Mathlib/GroupTheory/Subsemigroup/Center.lean

```diff
@@ -23,92 +23,216 @@ import Mathlib.Data.Int.Cast.Lemmas
 
 We provide `Submonoid.center`, `AddSubmonoid.center`, `Subgroup.center`, `AddSubgroup.center`,
 `Subsemiring.center`, and `Subring.center` in other files.
+
+## References
+
+* [Cabrera García and Rodríguez Palacios, Non-associative normed algebras. Volume 1]
+  [cabreragarciarodriguezpalacios2014]
 -/
 
 
 variable {M : Type*}
 
 namespace Set
 
-variable (M)
+/-- Conditions for an element to be additively central -/
+structure IsAddCentral [Add M] (z : M) : Prop where
+  /-- addition commutes -/
+  comm (a : M) : z + a = a + z
+  /-- associative property for left addition -/
+  left_assoc (b c : M) : z + (b + c) = (z + b) + c
+  /-- middle associative addition property -/
+  mid_assoc (a c : M) : (a + z) + c = a + (z + c)
+  /-- associative property for right addition -/
+  right_assoc (a b : M) : (a + b) + z = a + (b + z)
+
+/-- Conditions for an element to be multiplicatively central -/
+@[to_additive]
+structure IsMulCentral [Mul M] (z : M) : Prop where
+  /-- multiplication commutes -/
+  comm (a : M) : z * a = a * z
+  /-- associative property for left multiplication -/
+  left_assoc (b c : M) : z * (b * c) = (z * b) * c
+  /-- middle associative multiplication property -/
+  mid_assoc (a c : M) : (a * z) * c = a * (z * c)
+  /-- associative property for right multiplication -/
+  right_assoc (a b : M) : (a * b) * z = a * (b * z)
+
```

Comment:
Can you add the theorem `IsMulCentral.left_comm` and `.right_comm` here to match `Commute.right_comm`? I think it will shorten some proofs.

Reply:
Like this? https://github.com/leanprover-community/mathlib4/pull/6996/commits/1ae3734783e49a1d70a9687196d0775c85136c54